hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/integration/CompactFormatSplitBrainTest.java

```diff
@@ -81,108 +86,136 @@ public void teardown() {
-        assertQueryForNodeMapUsing(instance1);
+        assertQueryForEmployeeMapUsing(smallerBrain.getRandomInstance());
+        assertQueryForNodeMapUsing(largerBrain.getRandomInstance());
     }
 
-    private void splitCluster() {
-        for (HazelcastInstance splitAInstance : splitA) {
-            for (HazelcastInstance splitBInstance : splitB) {
-                SplitBrainTestSupport.blockCommunicationBetween(splitAInstance, splitBInstance);
-                closeConnectionBetween(splitAInstance, splitBInstance);
+    private static class Brain {
+        private final List<HazelcastInstance> instances;
+        private final Random random = new Random();
+
+        private Brain(List<HazelcastInstance> instances) {
+            this.instances = instances;
+        }
+
+        public HazelcastInstance getRandomInstance() {
+            return instances.get(random.nextInt(instances.size()));
+        }
+
+        public void splitFrom(Brain other) {
+            List<HazelcastInstance> otherBrainInstances = other.instances;
+            for (HazelcastInstance instance : instances) {
+                for (HazelcastInstance otherBrainInstance : otherBrainInstances) {
+                    blockCommunicationBetween(instance, otherBrainInstance);
+                }
+            }
+
+            for (HazelcastInstance instance : instances) {
+                for (HazelcastInstance otherBrainInstance : otherBrainInstances) {
+                    closeConnectionBetween(instance, otherBrainInstance);
+                }
+            }
+        }
+
+        public void mergeWith(Brain other) {
+            for (HazelcastInstance instance : instances) {
+                for (HazelcastInstance otherBrainInstance : other.instances) {
+                    unblockCommunicationBetween(instance, otherBrainInstance);
+                }
             }
         }
+    }
+
+    private void splitCluster() {
+        largerBrain.splitFrom(smallerBrain);
 
-        // make sure that cluster is split as [1 , 2, 3] , [4, 5]
-        assertClusterSizeEventually(3, instance1, instance2, instance3);
-        assertClusterSizeEventually(2, instance4, instance5);
+        // make sure that cluster is split
+        assertClusterSizeEventually(largerBrain.instances.size(), largerBrain.instances);
+        assertClusterSizeEventually(smallerBrain.instances.size(), smallerBrain.instances);
     }
 
     private void mergeCluster() {
-        for (HazelcastInstance splitAInstance : splitA) {
-            for (HazelcastInstance splitBInstance : splitB) {
-                SplitBrainTestSupport.unblockCommunicationBetween(splitAInstance, splitBInstance);
-            }
-        }
+        largerBrain.mergeWith(smallerBrain);
 
-        assertClusterSizeEventually(5, instance1, instance2, instance3, instance4, instance5);
+        assertClusterSizeEventually(allInstances.size(), allInstances);
+        waitAllForSafeState(allInstances);
```

Comment:
Why did you add this? Why not add it to splitCluster as well?

Reply:
I saw that it is being used in the `SplitBrainTestSupport`. Let me add it there as well